processes/water_quality.py

```diff
@@ -0,0 +1,178 @@
+import xarray as xr
+import dask
+import utils
+from os import path
+
+from cubequery.tasks import CubeQueryTask, Parameter, DType
+from datacube_utilities import import_export
+from datacube_utilities.masking import mask_good_quality
+from datacube_utilities.dc_water_quality import tsm
+
+
+class WaterQuality(CubeQueryTask):
+    """
+    Task for calculation of Total Suspended Matter(TSM) in water bodies. Uses inputs of
+    Landsat-4, Landsat-5, Landsat-7, Landsat-8 and Sentinel-2.
+
+    TSM stands for "Total Suspended Matter" - also called TSS which stands for "Total Suspended
+    Solids". It is the dry-weight of particles suspended (not dissolved) in a body of water. It is a
+    proxy of water quality.
+
+    This task is based from work by ceos-seo at the following link,
+    https://github.com/ceos-seo/data_cube_notebooks/blob/master/TSM_Demo_Notebook.ipynb, where the
+    TSM equation originates from.
+
+    Academic papers include Lymburner, L., Botha, E., Hestir, E., Anstee, J., Sagar, S., Dekker, A.
+    and Malthus, T., 2016. Landsat 8: providing continuity and increased precision for measuring
+    multi-decadal time series of total suspended matter. Remote Sensing of Environment, 185,
+    pp.108-118.
+
+    This paper demonstrates continuity between the Landsat sensors for TSM assessment. Calibration
+    for S2 use has not been carried out.
+    """
+
+    display_name = "Water Quality"
+    description = "Water Quality, showing the Total Suspeded Matter in water bodies."
+
+    parameters = [
+        Parameter("aoi", "AOI", DType.WKT, "Area of interest"),

```

Comment:
This is mostly a personal OCD, but it would make sense to keep consistency on parameter's description ending in '.' or not, since this will be used in the front end as it is.